src/shared/ErrorBoundaryBase.jsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import logger from '../utils/logger';
+
+const propTypes = {
+  /**
+   * Callback function that will be executed when an error is caught by the
+   * error boundary. The caught error will be provided as a function parameter.
+   * Ex. onCatchError(caughtError)
+   */
+  onCatchError: PropTypes.func.isRequired,
+  /**
+   * Components to render within the context of the error boundary. Exceptions
+   * thrown by these components during their render lifecycle will be caught by
+   * the error boundary.
+   *
+   * If an error is caught by the error boundary, the provided children will not
+   * be rendered.
+   */
+  children: PropTypes.node,
+};
+
+/**
+ * The ErrorBoundaryBase component provides a basic error boundary
+ * implementation with callback props allowing for customized error handling
+ * workflows.
+ */
+class ErrorBoundaryBase extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { error: undefined };
+  }
+
+  static getDerivedStateFromError(error) {
+    return { error };
+  }
+
+  componentDidCatch(error) {
+    logger.error(error);
```

Comment:
are there any scenarios where we don't want to log the error?

Reply:
That's a good question. Given the use case for this error boundary (e.g. handling egregious fatal errors), I honestly don't see a scenario where we would want to suppress that error from the logs. And I wouldn't expect consumers to use this component to implement "soft" error handling (something that would typically be handled with try/catch).